### PR TITLE
feat: add IncomingRequest::getRawInputVar() method

### PR DESCRIPTION
**Pull request by michalsn**
**Description**
Working with raw input data should be as easy as in other cases. `getRawInputVar()` adds missing features that are available for other methods, like retrieving specified indexes and filtering data.

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by kenjis** [comment]
I have a question: https://github.com/codeigniter4/CodeIgniter4/pull/6910

File: user_guide_src/source/incoming/incomingrequest/039.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+// foo=one&bar=two&baz[]=10&baz[]=20
```
Comment (kenjis) [inline user_guide_src/source/incoming/incomingrequest/039.php:3]: Is this better?
```suggestion
// When the request body is 'foo=one&bar=two&baz[]=10&baz[]=20'
```

File: user_guide_src/source/incoming/incomingrequest/039.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+var_dump($request->getRawInputVar('bar'));
+
```
Comment (kenjis) [inline user_guide_src/source/incoming/incomingrequest/039.php:5]: Remove this empty line.

File: user_guide_src/source/incoming/incomingrequest/039.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+// Outputs: two
+
+// foo=one&bar=two&baz[]=10&baz[]=20
+var_dump($request->getRawInputVar(['foo', 'bar']));
+
```
Comment (kenjis) [inline user_guide_src/source/incoming/incomingrequest/039.php:10]: Remove this empty line.

File: user_guide_src/source/incoming/incomingrequest/039.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ * Outputs:
+ * [
+ *      'foo' => 'one',
+ *      'bar' => 'two'
+ * ]
+ */
+
+// foo=one&bar=two&baz[]=10&baz[]=20
+var_dump($request->getRawInputVar('baz'));
+
```
Comment (kenjis) [inline user_guide_src/source/incoming/incomingrequest/039.php:21]: Remove this empty line.

File: user_guide_src/source/incoming/incomingrequest/039.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+ * Outputs:
+ * [
+ *      10,
+ *      20
+ * ]
+ */
+
+// foo=one&bar=two&baz[]=10&baz[]=20
+var_dump($request->getRawInputVar('baz.0'));
+
```
Comment (kenjis) [inline user_guide_src/source/incoming/incomingrequest/039.php:32]: Remove this empty line.

File: user_guide_src/source/incoming/incomingrequest/039.php [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+// foo=one&bar=two&baz[]=10&baz[]=20
+var_dump($request->getRawInputVar('baz'));
+
+/*
+ * Outputs:
+ * [
+ *      10,
+ *      20
```
Comment (kenjis) [inline user_guide_src/source/incoming/incomingrequest/039.php:26]: Aren't these strings?

```suggestion
 *      '10',
 *      '20'
```

**Comment by MGatner** [comment]
Kenjis hit this hard. Ping me if you need another review.

**Review by kenjis — APPROVED** [review]

**Review by MGatner — APPROVED** [review]

**Pull request merged**